python/tank/platform/validation.py

```diff
@@ -83,20 +83,22 @@ def validate_platform(descriptor):
             raise TankError("The current operating system '%s' is not supported."
                             "Supported platforms are: %s" % (nice_system_name, supported_platforms))
 
-    
-def get_missing_frameworks(descriptor, environment):
+
+def get_missing_frameworks(descriptor, environment, yml_file):
     """
     Returns a list of framework descriptors by the given descriptor required but not present 
     in the given environment.
-    
+
     returns items on the following form:
     [{'version': 'v0.1.0', 'name': 'tk-framework-widget'}]
-    
+
     :returns: list dictionaries, each with a name and a version key.
     """
     required_frameworks = descriptor.get_required_frameworks()
-    current_framework_instances = environment.get_frameworks()
-    
+    current_framework_instances = [
+        fw for fw in environment.get_frameworks() if environment.is_framework_available_from(fw, yml_file)
```

Comment:
I agree. It totally makes sense. The code ended up the way it is currently because originally I was using two methods that were existing on the environment object, but now I had to introduce a new one, so it makes sense to just make one method that does it all. Worst case scenario we end up exposing `is_framework_available` in the future.